### DISCUSSION TITLE
Usage Example / Question

### DISCUSSION THREAD

**Discussion by nashford77** [gravitl/netmaker#1200] [General]
Can this be used to do corporate client based VPN (multiple or one even, server config's & many client configs) - if so, can you change keys via API as needed & do you support static keys as well?

Where do the client config's get pulled / how do they get dumped for an admin to issue them out?

Also, can I mesh backends (connect multiple Datacenters) with multiple routable frontends (IE each DC has unique client configs (or shared) and connecting to any one can route me to any resource in the mesh.

 Usage would be for high 100's of thousand users potentially - will this scale networking wise?

**Comment by mattkasun** [comment]
> Can this be used to do corporate client based VPN (multiple or one even, server config's & many client configs) - if so, can you change keys via API as needed & do you support static keys as well?
> 
keys  rotation can be initiated through the  UI or api call

> Where do the client config's get pulled / how do they get dumped for an admin to issue them out?
> 
client configs are distributed to the netclient agent running on nodes.

> Also, can I mesh backends (connect multiple Datacenters) with multiple routable frontends (IE each DC has unique client configs (or shared) and connecting to any one can route me to any resource in the mesh.
yes
> 
> Usage would be for high 100's of thousand users potentially - will this scale networking wise?
yes
